### DISCUSSION TITLE
Background difficulty calculation breaks difficulty recommendation

### DISCUSSION THREAD

**Discussion by LittleEndu** [ppy/osu#19424] [Q&A]
At the point where new beatmap is selected that hasn't had its difficulty calculated, the star rating is -1 and difficulty recommender will select the first beatmap.

![debugging](https://user-images.githubusercontent.com/11320483/181349714-b45be65b-64ec-419e-b851-86d14ee6fcff.png)

**Comment by frenzibyte** [comment]
I don't really see this as an issue because the entirety of the beatmap would display no star rating. If anything, it should fall back to the online star rating.

**Reply by peppy** [reply]
i also think this is fine

**Reply by LittleEndu** [reply]
Yeah I guess this is fine because calculation of all difficulties is done only once. Just noticed this because I have 4k+ sets in lazer, it took quite a while to calculate them all. 

I didn't test if this also breaks on newly downloaded beatmaps because if so, a fallback to online star rating would be good to have.
